### PR TITLE
NYT-674: support auth

### PR DESCRIPTION
**Pull request by dmi-feo**

### CODE REVIEW

**Comment by dmi-feo** [comment]
DO NOT MERGE until we figure out what the heck is going on with yt images

File: ytsaurus.go [hunk 1]
```diff
@@ -65,6 +65,19 @@ func (y *YTsaurusContainer) NewClient(ctx context.Context) (yt.Client, error) {
 	return client, nil
 }
 
+// WithAuth enables authentication on http proxies and creates `admin` user with password and token `password`.
```
Comment (l0kix2) [inline ytsaurus.go:68]: Maybe we could provide constants for password/token here in lib so we don't have to hardcode them in the user code.

File: ytsaurus_test.go [hunk 2]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"testing"
 
+	"go.ytsaurus.tech/yt/go/yterrors"
+
```
Comment (l0kix2) [inline ytsaurus_test.go:8]: hm, linter doesn't catch unsorted imports 🤔

**Review by l0kix2 — CHANGES_REQUESTED** [review]

File: ytsaurus_test.go (hunk 2 above)
Comment (dmi-feo) [reply ytsaurus_test.go:8]: It seems that goimports is happy with this sorting, which is strange

**Comment by dmi-feo** [comment]
The fail of the test is expected since `stable` version of yt image has been rolled back

**Comment by l0kix2** [comment]
> The fail of the test is expected since `stable` version of yt image has been rolled back

You mean ytlocal stable? As far as i saw issue was localized. Will it be back or we better use some nightly ytlocal version here?

**Comment by dmi-feo** [comment]
> As far as i saw issue was localized.

Probably. They say they got something related to ipv4/ipv6, but I don't know what exactly it is. However, stable image is still too old for this rock'n'roll.

> we better use some nightly ytlocal version here?

That's a nice idea since only nightly builds have versioning :)

**Review by l0kix2 — APPROVED** [review]

**Comment by l0kix2** [comment]
Great, thanks!

**Comment by dmi-feo** [comment]
So, we can either
- merge it with the nightly image
- or wait for stable release update

The first approach has its own advantages (all users of this lib will use the same image) and disadvantages (we can't be sure everything works fine in this nightly buid)

**Comment by dmi-feo** [comment]
Oh, it's merged)

**Pull request merged**